tests/system/providers/amazon/aws/example_boto3.py

```diff
@@ -0,0 +1,138 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
A lot of comments are related to how AWS write their system tests and not necessarily flaws in your code. I think a good way to process these comments is to take another AWS system test as comparison (e.g. [this one](https://github.com/apache/airflow/blob/main/tests/system/providers/amazon/aws/example_local_to_s3.py)). But overall it looks good!